internal/user/repository/eventsourcing/eventstore.go

```diff
@@ -0,0 +1,760 @@
+	if err != nil {
+		return err
+	}
+	phoneCode.Code = phoneCodeCrypto
+	phoneCode.Expiry = es.PhoneVerificationCode.Expiry()
+
+	updateAggregate := PhoneVerificationCodeAggregate(es.AggregateCreator(), repoExisting, phoneCode)
+	err = es_sdk.Push(ctx, es.PushAggregates, repoExisting.AppendEvents, updateAggregate)
+	if err != nil {
+		return err
+	}
+
+	es.userCache.cacheUser(repoExisting)
+	return nil
+}
+
+func (es *UserEventstore) AddressByID(ctx context.Context, userID string) (*usr_model.Address, error) {
+	if userID == "" {
+		return nil, caos_errs.ThrowPreconditionFailed(nil, "EVENT-di8ws", "userID missing")
+	}
+	user, err := es.UserByID(ctx, userID)
+	if err != nil {
+		return nil, err
+	}
+
+	if user.Address != nil {
+		return user.Address, nil
+	}
+	return nil, caos_errs.ThrowNotFound(nil, "EVENT-so9wa", "address not found")
+}
+
+func (es *UserEventstore) ChangeAddress(ctx context.Context, address *usr_model.Address) (*usr_model.Address, error) {
+	existing, err := es.UserByID(ctx, address.AggregateID)
+	if err != nil {
+		return nil, err
+	}
+	repoExisting := model.UserFromModel(existing)
+	repoNew := model.AddressFromModel(address)
+
+	updateAggregate := AddressChangeAggregate(es.AggregateCreator(), repoExisting, repoNew)
+	err = es_sdk.Push(ctx, es.PushAggregates, repoExisting.AppendEvents, updateAggregate)
+	if err != nil {
+		return nil, err
+	}
+
+	es.userCache.cacheUser(repoExisting)
+	return model.AddressToModel(repoExisting.Address), nil
+}
+
+func (es *UserEventstore) OTPByID(ctx context.Context, userID string) (*usr_model.OTP, error) {
+	if userID == "" {
+		return nil, caos_errs.ThrowPreconditionFailed(nil, "EVENT-do9se", "userID missing")
+	}
+	user, err := es.UserByID(ctx, userID)
+	if err != nil {
+		return nil, err
+	}
+
+	if user.OTP != nil {
+		return user.OTP, nil
+	}
+	return nil, caos_errs.ThrowNotFound(nil, "EVENT-dps09", "otp not found")
+}
+
+func (es *UserEventstore) AddOTP(ctx context.Context, userID string) (*usr_model.OTP, error) {
+	existing, err := es.UserByID(ctx, userID)
+	if err != nil {
+		return nil, err
+	}
+	if existing.OTP != nil && existing.OTP.State == usr_model.MFASTATE_READY {
+		return nil, caos_errs.ThrowAlreadyExists(nil, "EVENT-do9se", "user has already configured otp")
+	}
+	key, err := totp.Generate(totp.GenerateOpts{Issuer: es.Multifactors.OTP.Issuer, AccountName: userID})
+	if err != nil {
+		return nil, err
+	}
+	encrypted, err := crypto.Encrypt([]byte(key.Secret()), es.Multifactors.OTP.CryptoMFA)
+	if err != nil {
+		return nil, err
+	}
+	repoOtp := &model.OTP{Secret: encrypted}
+	repoExisting := model.UserFromModel(existing)
+	updateAggregate := MfaOTPAddAggregate(es.AggregateCreator(), repoExisting, repoOtp)
+	err = es_sdk.Push(ctx, es.PushAggregates, repoExisting.AppendEvents, updateAggregate)
+	if err != nil {
+		return nil, err
+	}
+
+	es.userCache.cacheUser(repoExisting)
+	otp := model.OTPToModel(repoExisting.OTP)
+	otp.Url = key.URL()
+	otp.SecretString = key.Secret()
```

Comment:
maybe passing

Reply:
i think this would be wrong because we dont need it always in otptomodel

Reply:
okay